src/Microsoft.CodeAnalysis.Razor.Workspaces/WorkspaceMonitor.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Razor
+{
+    public class WorkspaceMonitor
+    {
+        private readonly Workspace _workspace;
+
+        public WorkspaceMonitor(Workspace workspace)
```

Comment:
Didn't think it was worth following the abstract class => static create method pattern due to how this class will be used. Let me know if you think differently.